lib/super_tooltip.dart

```diff
@@ -999,6 +1015,32 @@ EdgeInsets _getTooltipMargin({
   }
 }
 
+EdgeInsets _getTooltipPadding({
+  @required ShowCloseButton showCloseButton,
+  @required double closeButtonSize,
+  @required PreferredDirection preferredDirection,
+}) {
+  final top =
+      (showCloseButton == ShowCloseButton.inside) ? closeButtonSize : 0.0;
+
+  switch (preferredDirection) {
+    case PreferredDirection.down:
+      return EdgeInsets.only(top: top);
+
+    case PreferredDirection.up:
+      return EdgeInsets.only(top: top);
+
+    case PreferredDirection.left:
+      return EdgeInsets.only(top: top);
+
+    case PreferredDirection.right:
+      return EdgeInsets.only(top: top);
+
+    default:
+      throw ArgumentError(preferredDirection);
+  }
```

Comment:
Why do you return the same in  all cases?

Reply:
I wanted to give a top padding in all 4 directions

Reply:
yes, but if its the same in all direction, what is the `switch` then good for?

Reply:
True, will need to update it and get rid of `switch`